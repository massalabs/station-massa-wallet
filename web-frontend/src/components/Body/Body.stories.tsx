import Body, { BodyProps } from './Body';

export default {
  component: Body,
  title: 'Body',
};

const Template = (args: BodyProps) => <Body {...args} />;

export const DefaultBody = () => (
  <Template>
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc consequat
    aliquam felis, ac vestibulum nisl placerat in. In hac habitasse platea
    dictumst. Pellentesque convallis erat eu enim tincidunt, a consectetur elit
    efficitur.
  </Template>
);
