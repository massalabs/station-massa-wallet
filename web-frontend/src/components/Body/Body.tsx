import React from 'react';

export interface BodyProps {
  children: React.ReactNode;
}

// TODO: Poppins font family

export default function Body(props: BodyProps) {
  return (
    <p className="text-base font-medium text-black leading-6 tracking-massa">
      {props.children}
    </p>
  );
}
