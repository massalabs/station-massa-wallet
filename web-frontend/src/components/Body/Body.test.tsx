import renderer from 'react-test-renderer';
import { cleanup } from '@testing-library/react';
import Body from './Body';

afterEach(cleanup);

describe('Body', () => {
  it('should render a body with the given text', () => {
    const message = `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc consequat aliquam felis,
			ac vestibulum nisl
			placerat in. In hac habitasse platea dictumst. Pellentesque convallis erat eu enim tincidunt, a consectetur
			elit efficitur.`;
    const body = <Body>{message}</Body>;

    // snapshot test
    const componentSnapchat = renderer.create(body);
    const tree = componentSnapchat.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
