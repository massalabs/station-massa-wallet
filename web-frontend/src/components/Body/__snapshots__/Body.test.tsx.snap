// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Body should render a body with the given text 1`] = `
<p
  className="text-base font-medium text-black leading-6 tracking-massa"
>
  Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nunc consequat aliquam felis,
			ac vestibulum nisl
			placerat in. In hac habitasse platea dictumst. Pellentesque convallis erat eu enim tincidunt, a consectetur
			elit efficitur.
</p>
`;
